### DISCUSSION TITLE
Support for explicit declaration of the language of a post

### DISCUSSION THREAD

**Discussion by venteto** [bluesky-social/atproto#2027] [Bluesky Lexicons]
Especially with custom feeds being all the proposed rage now, it would be useful if the protocol implemented an attribute where each post, individually, declared its language. This is even if the person writing the post did not bother to manually set the attribute and for example the client app guessed the language based on the words present in the post. The client app could also perhaps have a global account setting for a default language, as well as an explicit widget in a new post creation UI for overriding that default / setting the post language explicitly.

This would enable custom feed algorithms where you could include accounts which often post in multiple languages at different times, but include posts from that account in the custom feed _only if_ they are in a desired language.

Like for example, you have an account that you'd like to include in an Enviromental feed, but only when their posts are in English, not when they are in Italian.

This would perhaps matter less though if the client app automatically translated all posts not in the default language of the account which is reading the posts.

**Comment by DavidBuchanan314** [comment]
This is already a feature: https://github.com/bluesky-social/atproto/blob/de2dbc2903a383f092756d3d2ad1ff3753361e76/lexicons/app/bsky/feed/post.json#L33-L37

**Reply by venteto** [reply]
That's good but does it rely completely on the user accurately (and manually) declaring an language on their post?

I mean if they have English selected but write the post in French, as far as the entire network is concerned is it going to appear as an English post? Or will the client app catch the error and assign French during the posting process?

**Reply by pfrazee** [reply]
No but "catching the wrong language" during composition is something I'd like to add. Actually going to open an issue for that.
